compiler/rustc_codegen_gcc/src/asm.rs

```diff
@@ -546,9 +546,26 @@ impl<'a, 'gcc, 'tcx> AsmBuilderMethods<'tcx> for Builder<'a, 'gcc, 'tcx> {
         }
 
         if !options.contains(InlineAsmOptions::PRESERVES_FLAGS) {
-            // TODO(@Commeownist): I'm not 100% sure this one clobber is sufficient
-            // on all architectures. For instance, what about FP stack?
-            extended_asm.add_clobber("cc");
+            match asm_arch {
+                InlineAsmArch::PowerPC | InlineAsmArch::PowerPC64 => {
+                    let clobbers = ["cr0", "xer"];
```

Comment:
this should probably include `cr1` for the FP `.` instructions and `cr6` for the vector instructions. The rest of the `cr` fields should need explicit instruction arguments so are much less likely to be forgotten.